engine/src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -40,6 +40,7 @@ case class WorkflowDescriptor(id: WorkflowId,
                               backend: Backend,
                               configCallCaching: Boolean,
                               lookupDockerHash: Boolean,
+                              configWorkflowFailureMode: WorkflowFailureMode,
```

Comment:
I think that if this is going to be an attribute of the class it should be the final definitive value.
Now it is used as a default when computing the actual value on line 65.
The full logic of determining the mode should be done int eh apply IMO. It would avoid having `configWorkflowFailureMode` and `workflowFailureMode` where really you only care about the latter.